pallene/to_ir.lua

```diff
@@ -38,15 +38,26 @@ function ToIR:init()
     self.fun_id_of_exp  = {} -- { ast.Exp  => integer }
     self.glb_id_of_decl = {} -- { ast.Decl => integer } -- non-exported globals
     self.glb_id_of_var  = {} -- { ast.Var  => integer } -- exported global
+    self.func_stack     = {} -- list of function metadata
 end
 
 function ToIR:enter_function(f_id)
     -- Function-specific variables
     -- These are re-initialized each time.
-    self.func = self.module.functions[f_id]
-    self.loc_id_of_decl = {} -- { ast.Decl => integer }
-    self.call_exps      = {} -- { ast.Exp.CallFunc }
-    self.dsts_of_call   = {} -- { ast.Exp => { var_id } }
+    local func_data = {
+        loc_id_of_decl = {}, -- { ast.Decl => integer }
+        call_exps      = {}, -- { ast.Exp.CallFunc }
+        dsts_of_call   = {}, -- { ast.Exp => { var_id } }
+        func           = self.module.functions[f_id],
+        f_id           = f_id
+    }
```

Comment:
Ah yeah. I was also wondering it it can use a `tag`?
Also, where would this live exactly? An `ir.FuncData` probably won't make a lot of sense .

Reply:
I'm not sure if it needs a tag, because we'll never be using it in an "if elseif".

I agree that it's a bit strange that the way to create a tagged datatype is different from an untagged one. But that's how things go around here...

> Also, where would this live exactly?

It can live in the `to_ir`. I was just using the `ir.Function` as an example because in the ir.lua we can use that name in the comments.

> ir.FuncData probably won't make a lot of sense .

"Data" is a lousy name. I'm OK with it but if you can come up with a more self-descriptive name that would be great :)

Reply:
`FuncInfo` or `FuncMetaData` is probably the best I can do 😅 .
Out of the three, I would pick `FuncMetaData`.

Reply:
Let's go with `FuncInfo` then. I don't like "metadata" because in my head that sounds like it is immutable data.